### ISSUE TITLE
After the update 5.5.5 -> 5.6, scrolling from the mouse wheel stopped working normally in chats - it works in jerks, skips events from the mouse wheel. Rolled back to 5.5.5 - it works fine there.

### ISSUE THREAD

**Issue by conductor111** [telegramdesktop/tdesktop#28498]
### Steps to reproduce

Update from 5.5.5 -> 5.6.

### Expected behaviour

Scrolling from the mouse wheel should work smoothly in chats.

### Actual behaviour

Scrolling from the mouse wheel works in jerks, skips events from the mouse wheel.

### Operating system

Debian 11 x86-64

### Version of Telegram Desktop

5.6

### Installation source

Static binary from official website

**Comment by Remu-rin** [comment]
Same for me in Kubuntu 22.04, X11. Started from 5.5.6 beta version. Last version without the bug is 5.5.5.

Looks like it is related to moving mouse from one scrollable area to another (e.g. chat and chatlist, or chat and sidebar), and after that it ignores first 1-2 events from mouse wheel and jumps to 3 positions at once. Then it works normally for 1 event up/down, until you move mouse to another area and back, which again triggers the problem.

**Comment by pekmop1024** [comment]
Confirming same on Arch Linux x86_64 (X11. XFCE). Rolled back to 5.5.5 for now, it's unusable and annoying.

**Comment by All-The-Foxes** [comment]
Same issue on Ubuntu 24.04.1 LTS

**Comment by gromov-io** [comment]
Same issue on Ubuntu 24.04

**Comment by Tynach** [comment]
There are a few nuances to this behavior I'd like to document, just in case it helps narrow down the problem:

1. Mouse movement seems to cause it. If the mouse stays perfectly still, the scrolling behaves just fine.. Until the mouse moves again. After the mouse moves, the problem occurs again. Even if the mouse is still while the scroll wheel is being used, if the mouse has moved since the last time the scroll wheel was used, the problem occurs.
2. The mouse wheel events seem to actually be processed, but the effects of the events don't seem to occur immediately. If I open the media viewer, move the mouse, stop moving the mouse, scroll one tick up (go to the previous image), then scroll another single tick up, the media viewer jumps to 2 images ago. This indicates that at least the media viewer consistently keeps track of which image you're supposed to be seeing, but doesn't actually change which image is being shown.
3. Other areas like the chat list and the main area showing chats, are less consistent about this. Moving the mouse sometimes causes this issue, sometimes doesn't. Sometimes the issue happens even without moving the mouse, or at least seems to, but other times requires mouse movement.

All of this happens in both 5.6.0 *and* 5.6.1.

**Comment by ilya-fedin** [comment]
I bet it's just a Qt regression as always. Nothing else colud cause such werid issue, especially only on one platform. Personally I don't have a mouse wheel, only a touchpas and this seems to be unreproducible with touchpad.

**Comment by Tynach** [comment]
@ilya-fedin I know that in some programs, the first mouse wheel event given to the program will be eaten if the program uses libinput. But since the events seem to be kept track of, just in a weird hidden sort of way, I'm not really sure that's what's happening here.

Still, makes me wonder if Qt decided to do something funky regarding libinput?

**Comment by ilya-fedin** [comment]
Well, client applications don't use libinput to my knowledge. It's either a change in how Qt uses X11 API or how Qt uses Wayland API (depending on where it happens).

**Comment by Tynach** [comment]
Libinput is a newer method of handling input, that's usable from both Wayland *and* X11. Its main goal was to help with smooth scrolling with touchpads, which is your use case... So if suddenly the scrolling behavior got a lot better for you, I would not be surprised.

**Comment by ilya-fedin** [comment]
> Libinput is a newer method of handling input, that's usable from both Wayland _and_ X11.

No, that's a widespread confusion. It's a wrapper on top of evdev, to be used by the Xorg server and Wayland compositors to avoid handling various low-level things. Clients just use X11/Wayland APIs.

**Comment by Tynach** [comment]
I googled 'wayland mouse events' and got a page detailing how to use libinput with Wayland.

**Comment by Tynach** [comment]
I never said it did, I said that programs in X11 and Wayland can use libinput, which is true. On X11, it's implemented as a generic input driver, that wraps the other drivers so that applications can use it on X11 similarly to how they do on Wayland.

On Wayland, it seems to be the only way for applications to get mouse pointer events.

**Comment by ilya-fedin** [comment]
> Still, makes me wonder if Qt decided to do something funky regarding libinput?

Well, you said this. But Qt doesn't use libinput APIs as I already said. tdesktop official binary doesn't even link to it.

**Comment by ilya-fedin** [comment]
> On Wayland, it seems to be the only way for applications to get mouse pointer events.

That's really an implementation detail. Clients don't know what compositors use under the hood. The compositor could parse evdev devices directly if they want to.

**Comment by ilya-fedin** [comment]
Qt uses libinput only when operating without Wayland or X11, in backends such as eglfs, vkkhrdisplay, linuxfb, directfb. tdesktop official binary doesn't link those backends so libinput is not linked either. gtk doesn't use libinput at all, you won't find it linking to libinput even in distro packages.

**Comment by Tynach** [comment]
All I know, is that in Firefox to solve the problem I had to set an environment variable at one time. I don't remember exact details, but I remember it being related to Firefox changing how they handled input to start using libinput.

**Comment by ilya-fedin** [comment]
Firefox doesn't link to libinput either so you have remembered what you did really poorly.

**Comment by Tynach** [comment]
Hm, that might be it. My memory is fuzzy on this and I might have confused XInput and Libinput.

**Comment by All-The-Foxes** [comment]
I am having the issue on X11. I don't use Wayland.

**Comment by Remu-rin** [comment]
Seems fixed in QT 6.8: https://bugreports.qt.io/browse/QTBUG-129509
Probably will be fixed in next tdesktop version.

**Comment by TogarUshindi** [comment]
Having the same issue. Mouse's scroll wheel function seems to catch about half of scrolling attempts, intermittently. Sometimes several attempts to make one scroll. Happens through browsing images or chats. Every other app scrolls just fine, so this is localized to the Telegram desktop application.

I'm using a Logitech mouse, which can either scroll freely or clicky. When I put it into free-scroll mode, I spin the wheel and it scrolls very jerkily, not smooth like normal. This seems to have started on the most recent update.

I'm on Ubuntu 24, x11, Telegram 5.6.1

**Comment by borodinalive** [comment]
Была та же проблема. Она стала сильно меньше после отключения опции "Использовать системные настройки прокси" и переключение на "Отключить прокси" (sic!)

**Comment by Remu-rin** [comment]
> Seems fixed in QT 6.8: https://bugreports.qt.io/browse/QTBUG-129509
> Probably will be fixed in the next tdesktop version.

Fixed in 5.6.2 version.
